dbms/src/Flash/Mpp/MPPTaskManager.cpp

```diff
@@ -293,6 +305,25 @@ String MPPTaskManager::toString()
     return res + ")";
 }
 
+std::pair<std::shared_ptr<ProcessListEntry>, String> MPPTaskManager::getOrCreateQueryProcessListEntry(const MPPQueryId & query_id, const ContextPtr & context)
+{
+    std::lock_guard lock(mu);
+    auto [query_set, abort_reason] = getQueryTaskSetWithoutLock(query_id);
+    if (!abort_reason.empty())
+        return {nullptr, abort_reason};
+    if (query_set == nullptr)
+        query_set = addMPPQueryTaskSet(query_id);
```

Comment:
https://github.com/pingcap/tiflash/blob/986d9514325378eb0b63a158658e1e8c9072f0de/dbms/src/Flash/Mpp/MPPTaskManager.cpp#L254, should this branch be removed? Since initProcessListEntry is done before registerTask

Reply:
Yes, you are right, I've updated the code.